gravitee-am-service/src/test/java/io/gravitee/am/service/PasswordPolicyServiceTest.java

```diff
@@ -267,6 +292,64 @@ public void shouldUpdate() {
+        observer.assertComplete();
+        observer.assertValue(passwordPolicy ->
+                passwordPolicy.getId().equals(existingPolicy.getId()) &&
+                        passwordPolicy.getCreatedAt().equals(existingPolicy.getCreatedAt()) &&
+                        !passwordPolicy.getUpdatedAt().equals(existingPolicy.getUpdatedAt()) &&
+                        passwordPolicy.getReferenceId().equals(DOMAIN_ID) &&
+                        passwordPolicy.getReferenceType().equals(ReferenceType.DOMAIN) &&
+                        passwordPolicy.getName().equals(updatePasswordPolicy.getName()) &&
+                        passwordPolicy.getDefaultPolicy().equals(existingPolicy.getDefaultPolicy()));
+
+        verify(auditService).report(ArgumentMatchers.argThat(builder -> builder.build(MAPPER).getOutcome().getStatus().equals(Status.SUCCESS)));
+        verify(eventService).create(ArgumentMatchers.argThat(evt -> evt.getType().equals(Type.PASSWORD_POLICY) && evt.getPayload().getAction().equals(Action.UPDATE)));
+    }
+
+    @Test
+    public void shouldUpdateNotDefaultPolicy() {
+        var existingPolicy = createPasswordPolicy();
+        existingPolicy.setDefaultPolicy(Boolean.FALSE);
+        var updatePasswordPolicy = new UpdatePasswordPolicy();
+        updatePasswordPolicy.setName(UUID.randomUUID().toString());
+        updatePasswordPolicy.setDefaultPolicy(Boolean.TRUE);
+
+        when(passwordPolicyRepository.findByReferenceAndId(any(), any(), any())).thenReturn(Maybe.just(existingPolicy));
+        when(passwordPolicyRepository.update(any())).thenAnswer(answer -> Single.just(answer.getArguments()[0]));
+        when(eventService.create(any())).thenReturn(Single.just(new Event()));
+
+        TestObserver<PasswordPolicy> observer = cut.update(ReferenceType.DOMAIN, DOMAIN_ID, existingPolicy.getId(), updatePasswordPolicy, principal).test();
+
+        observer.awaitDone(10, TimeUnit.SECONDS);
+        observer.assertComplete();
+        observer.assertValue(passwordPolicy ->
+                passwordPolicy.getId().equals(existingPolicy.getId()) &&
+                        passwordPolicy.getCreatedAt().equals(existingPolicy.getCreatedAt()) &&
+                        !passwordPolicy.getUpdatedAt().equals(existingPolicy.getUpdatedAt()) &&
+                        passwordPolicy.getReferenceId().equals(DOMAIN_ID) &&
+                        passwordPolicy.getReferenceType().equals(ReferenceType.DOMAIN) &&
+                        passwordPolicy.getName().equals(updatePasswordPolicy.getName()) &&
+                        passwordPolicy.getDefaultPolicy().equals(existingPolicy.getDefaultPolicy()));
+
+        verify(auditService).report(ArgumentMatchers.argThat(builder -> builder.build(MAPPER).getOutcome().getStatus().equals(Status.SUCCESS)));
+        verify(eventService).create(ArgumentMatchers.argThat(evt -> evt.getType().equals(Type.PASSWORD_POLICY) && evt.getPayload().getAction().equals(Action.UPDATE)));
+    }
+
```

Comment:
suggestion: could be nice to mututalize this test as the only difference is the boolean values used at the beginning to define the defaultPolicy. wdyt?

Reply:
I would stick with current solution for better readability